File: calificaciones.py
estudiantes = []

def agregar_o_actualizar_estudiante():
    nombre = input("Ingrese el nombre del estudiante: ").strip()
    materias = ["Matematicas", "Lengua", "Sociales", "Naturales"]
    
    estudiante = next((e for e in estudiantes if e[0] == nombre), None)

    if estudiante:
        print("El estudiante ya existe. Puedes actualizar sus notas.")
    else:
        estudiante = [nombre, []]
        estudiantes.append(estudiante)

    for materia in materias:
        try:
            nota = int(input(f"Ingrese la nota para {materia} (o -1 para no cambiar): "))
            if 0 <= nota <= 10:
                materia_existente = next((m for m in estudiante[1] if m[0] == materia), None)
                if materia_existente:
                    materia_existente[1] = nota
                else:
                    estudiante[1].append([materia, nota])
        except ValueError:
            print("Nota inválida.")

    print("Registro actualizado.")

def mostrar_estudiantes():
    print("Lista de estudiantes y sus calificaciones:")
    print(estudiantes)

while True:
    print("\n1. Agregar/Actualizar estudiante\n2. Mostrar estudiantes\n3. Salir")
    opcion = input("Seleccione una opción: ")
    if opcion == "1":
        agregar_o_actualizar_estudiante()
    elif opcion == "2":
        mostrar_estudiantes()
    elif opcion == "3":
        print("Saliendo...")
        break
    else:
        print("Opción inválida.")

File: calificaciones.js
const prompt = require("prompt-sync")({ sigint: true });

let estudiantes = [];

function agregarOActualizarEstudiante() {
    let nombre = prompt("Ingrese el nombre del estudiante: ").trim();
    let materias = ["Matematicas", "Lengua", "Sociales", "Naturales"];
    let estudiante = estudiantes.find(e => e[0] === nombre);

    if (estudiante) {
        console.log("El estudiante ya existe. Puedes actualizar sus notas.");
    } else {
        estudiante = [nombre, []];
        estudiantes.push(estudiante);
    }

    materias.forEach(materia => {
        let nota = parseInt(prompt(`Ingrese la nota para ${materia} (o -1 para no cambiar): `));
        if (nota >= 0 && nota <= 10) {
            let materiaExistente = estudiante[1].find(m => m[0] === materia);
            if (materiaExistente) {
                materiaExistente[1] = nota;
            } else {
                estudiante[1].push([materia, nota]);
            }
        }
    });

    console.log("Registro actualizado.");
}

function mostrarEstudiantes() {
    console.log("Lista de estudiantes y sus calificaciones:");
    console.log(estudiantes);
}

while (true) {
    console.log("\n1. Agregar/Actualizar estudiante\n2. Mostrar estudiantes\n3. Salir");
    let opcion = prompt("Seleccione una opción: ");
    if (opcion === "1") {
        agregarOActualizarEstudiante();
    } else if (opcion === "2") {
        mostrarEstudiantes();
    } else if (opcion === "3") {
        console.log("Saliendo...");
        break;
    } else {
        console.log("Opción inválida.");
    }
}